### PR TITLE
lib, programs: remove unused includes

### PR DESCRIPTION
**Pull request by gemesa**

### CODE REVIEW

File: programs/gzip.c [hunk 1]
```diff
@@ -32,14 +32,11 @@
 #include "prog_util.h"
 
 #include <errno.h>
-#include <sys/types.h>
 #include <sys/stat.h>
 #ifdef _WIN32
 #  include <sys/utime.h>
 #else
-#  include <sys/time.h>
```
Comment (ebiggers) [inline programs/gzip.c:40]: `sys/time.h` and `utime.h` are sometimes needed, see `restore_timestamps()`.

Comment (gemesa) [reply programs/gzip.c]: My bad, I fixed it. I think the other headers can be safely removed.

**Comment by ebiggers** [comment]
Can you please squash your two commits so that you aren't introducing a bug and then fixing it?  Thanks!

**Comment by ebiggers** [comment]
Merged, thanks.

**Pull request merged**